src/Serializer/AbstractItemNormalizer.php

```diff
@@ -299,6 +304,17 @@ private function denormalizeRelation(string $attributeName, PropertyMetadata $pr
         }
 
         if (!is_array($value)) {
+            // repeat the code so that IRIs keep working with the json format
+            if (true === $this->allowPlainIdentifiers && $this->itemDataProvider) {
```

Comment:
What is `$this->itemDataProvider === null` and `true === $this->allowPlainIdentifiers` ? Instead of silently not doing anything, should we throw an exception instead?

Reply:
We could, maybe a bit overkill, this dependency should be a hard dep, it's just nullable to avoid breaking.